packages/loopring_v3/contracts/impl/libexchange/ExchangeBalances.sol

```diff
@@ -72,32 +71,33 @@ library ExchangeBalances
         uint balanceItem = hashImpl(balance, index, tradeHistoryRoot, 0);
         uint _id = tokenID;
         for (uint depth = 0; depth < 6; depth++) {
+            uint base = depth << 1 + depth;
```

Comment:
This actually costs 1 gas more than just `* 3` (5 gas vs 2*3).

Verifying the Merkle paths costs ~1.5M gas because of Poseidon... so doesn't really matter in any case. Whatever is most readable.